packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1677,6 +1657,27 @@ function updateTypesFromOperation(
     }
     const access = getAccessOverride(context, operation) ?? "public";
     diagnostics.pipe(updateUsageOrAccessOfModel(context, access, sdkType));
+
+    // after complete of usage calculation for httpBody, check whether it has
+    // conflicting usage between multipart and regular body
+    if (
+      sdkType.kind === "model" &&
+      ((!multipartOperation && (sdkType.usage & UsageFlags.MultipartFormData) > 0) ||
+        (multipartOperation &&
+          (sdkType.usage & UsageFlags.MultipartFormData) > 0 &&
+          ((sdkType.usage & UsageFlags.Json) | (sdkType.usage & UsageFlags.Xml)) > 0))
```

Comment:
As discussed in https://github.com/Azure/typespec-azure/issues/1510#issuecomment-2345259187, we shall usage existing usage `xml/Json` to distinguish whether a model has conflicting usage between multipart operation and regular operation.
And we shall move the logic position after complete of usage calculation.

Reply:
can you clean up this logic so it's easier to read? i.e. can you make vars like `isXml = sdkType.usage & UsageFlags.Xml > 0` because right now it's really hard to follow what the logic is trying to do here

Reply:
@iscai-msft i've refined the logic, could you review again and help to merge?

Reply:
do we want this to merge into the release branch or into the main branch?

Reply:
main is ok.